crypto-core/src/test/java/com/palantir/crypto2/jmh/EncryptionBenchmark.java

```diff
@@ -55,54 +60,107 @@
         @Param({"1048576", "10485760", "104857600"})
         public int numBytes;
 
+        @Param
+        public WriteStrategy writeStrategy;
+
         public byte[] data;
 
         public KeyMaterial key;
 
         @SuppressWarnings("RegexpSinglelineJava")
         @Setup
-        public void setup() {
+        public void setup() throws IOException {
             data = new byte[numBytes];
             random.nextBytes(data);
             key = KeyMaterials.generateKeyMaterial("AES", 256, 16);
+            for (WriteStrategy strategy : WriteStrategy.values()) {
+                validateWriteStrategy(data, strategy);
+            }
+        }
+    }
+
+    private static void validateWriteStrategy(byte[] data, WriteStrategy strategy) throws IOException {
```

Comment:
should this move to unit test?

Reply:
Perhaps, I added it here as I was tweaking the benchmarks and wanted to make sure I wasn't reporting bogus values!

I opened a second pr which applies the chunking strategy to CipherOutputStream via our factory methods, I've added test coverage to that PR.

Reply:
Adding the PR link: #587

Reply:
@carterkozak do we want to remove the validation piece now as #587 merged with tests?

Reply:
I could go either way, I don't mind validating that all WriteStrategies which may be a superset of our core implementations produce equivalent results. If it turns out to make the benchmarks much more expensive, we can remove the validation step, however it only occurs once per benchmark setup, not per-iteration.